prometheus_juju_exporter/exporter.py

```diff
@@ -13,15 +13,30 @@ class ExporterDaemon:
     def __init__(self, debug=False):
         """Create new daemon and configure runtime environment."""
         self.config = Config().get_config()
-        self.logger = self.setup_logging(debug=debug)
-        self.logger.info("Parsed config: {}".format(self.config.config_dir()))
+        self.logger = get_logger(debug=debug)
+        self.logger.info("Parsed config: %s" % self.config.config_dir())
         self._registry = CollectorRegistry()
         self.collector = CollectorDaemon(self._registry, debug=debug)
         self.logger.debug("Exporter initialized")
+        self.validate_config_values()
 
-    def setup_logging(self, debug):
-        """Return the correct Logging instance based on debug option."""
-        return get_logger(debug=debug)
+    def validate_config_values(self):
```

Comment:
1. this function feels out of place here, it should rather be part of config.py
2. it looks like you're assuming self.port and self.interval actually exist in the config file. What if they don't? It would be good to a have a test case for an empty or malformed config file
3. while unable to offer a framework for custom validators, confuse does have a few options beyond `.get(int)`. You may try something like `.as_choice(list(range(0, 65536)))`

Reply:
Thanks for the suggestions. I will give these more thoughts and make improvement.

Reply:
These suggestions has been addressed in the latest commit.